src/OrchardCore/OrchardCore.ContentManagement.Display/ContentDisplay/ContentItemDisplayCoordinator.cs

```diff
@@ -90,6 +92,35 @@ public async Task BuildDisplayAsync(ContentItem contentItem, BuildDisplayContext
                         }
                     }
 
+                    var tempContext = context;
+
+                    // Create a custom ContentPart shape that will hold the fields for dynamic content part (not implicit parts)
+                    // This allows its fields to be grouped and templated 
+
+                    if (part.GetType() == typeof(ContentPart) && contentTypePartDefinition.PartDefinition.Name != contentTypePartDefinition.ContentTypeDefinition.Name)
+                    {
+                        var shapeType = context.DisplayType != "Detail" ? "ContentPart_" + context.DisplayType : "ContentPart";
+
+                        var shapeResult = new ShapeResult(shapeType, ctx => ctx.ShapeFactory.CreateAsync(shapeType, () => Task.FromResult<IShape>(new ZoneHolding(() => ctx.ShapeFactory.CreateAsync("Zone", Arguments.Empty)))));
```

Comment:
`ContentPartDisplayDriver<TPart>.Factory` needs to get called to perform the shape building logic that is common to all parts

Reply:
The problem is that the standard part shape alternates aren't getting added to the shape's metadata with this implementation

Reply:
I know. But in this case the list of standard shapes is very limited because:
1- There is a single shape, while the base factory handles multiple ones.
2- The shape name is constant, hence the only alternates to add are part name and display type.

However I can try to make it available internally to reuse the same code.

Reply:
I tried, but I can't get it to work correctly. It never displays the fall-back ContentPart template when there is no custom one if I try to reuse the code.

Reply:
@sebastienros the shape name isn't constant (it comes from the part name), and the part name is not constant.

According to the documentation there are multiple alternates that should be added that combine the part name or part type with a combination of other identifiers - surely [these](https://orchardcore.readthedocs.io/en/latest/OrchardCore.Modules/OrchardCore.Templates/README/#contenttype_displaytype__partname) are needed. I would certainly expect them.

There must be a way of getting the template binding to work. I thought I had it working with my implementation of `DefaultContentPartDisplayDriver`, but maybe I was adding the alternates too late, and instead they needed to be added before the standard ones?  I think that could be done by passing an `Func<IShape, Task> initializeAsync` lambda into the ContentPartDisplayDriver's Factory method.